api/docs/ldstex.dox

```diff
@@ -106,17 +106,26 @@ discussed below.
 
 For sequences where the load-exclusive is in a different block than the corresponding
 store-exclusive, there are multiple scenarios where DynamoRIO can insert loads and stores
-in between the two.  First, for linking two blocks together, if the blocks are too far
-apart for a single branch instruction to reach, DynamoRIO jumps through an exit stub
-which contains loads and stores (see \ref page_aarch64_far).  Second, for building
-traces, DynamoRIO unlinks each block before execution while discovering the sequence of
-blocks to use for the new traces.  It assumes that unlinking does not perturb the
-application.  However, the unlink path goes through a stub with memory operations and
-then through DynamoRIO system code with a large number of loads and stores.  Third,
-indirect branches load entries from a hashtable, so an indirect branch between a
-load-exclusive and store-exclusive, while perhaps unlikely to occur in most code, is
-another case where even without a tool adding additional instruction the core DynamoRIO
-engine has a potential problem.
+in between the two.  First, an unlinked block will execute many loads and stores on its
+path into DynamoRIO and inside DynamoRIO system code.  Thus, on first execution of a pair
+of split load-exclusive and store-exclusive blocks, the store-exclusive will likely fail
+due to all of the operations in between while building its block.  Even once the two
+blocks are linked, though, if the blocks are too far apart for a single branch
+instruction to reach, DynamoRIO jumps through an exit stub which contains loads and
+stores in order to reach the target block (see \ref page_aarch64_far).
+
+The operations after an unlinked block also affect trace building.  During trace
```

Comment:
How about "... operations after unlinking a block also affect ... "? "unlinked block" may also mean a block that is simply not linked (like in the previous paragraph).

Reply:
Not following: this is referring to any unlinked block.  "operations after unlinking a block" sounds like either events happening inside the agent that did the unlocking rather than the block that got unlocked, or it implies events that only happen once when a block is marked unlinked, rather than every time an unlinked bock is executed.

Reply:
> rather than the block that got unlocked

I agree that this is what we want to say. I only was suggesting that we should make it clear that we're talking about a block that got unlinked now, rather than an "unlinked block" which may also mean a block that was simply never linked (like the usage in the previous paragraph).

Reply:
But this lead-in sentence is referring to the previous paragraph's discussion about blocks in an unlinked state in general, and even for the subsequent trace building discussion it could include a block that was just created and not yet linked, if trace discovery goes onto a new path.

Not sure it's worth further time...inclined to leave it and move on if that's ok.

Reply:
Sounds good